core/src/main/java/org/chodavarapu/datamill/http/Client.java

```diff
@@ -61,108 +75,155 @@ protected URLConnection createConnection(String uri) throws IOException {
         if (uriParameters != null && uriParameters.size() > 0) {
             uri = uriBuilder.build(uri, uriParameters);
         }
+        URIBuilder uriBuilder = new URIBuilder();
+
+
+        int firstSlash = uri.indexOf('/');
```

Comment:
Not sure I understand why you  are doing all the string manipulation here - I think just create a URI straight from the string and let the URI implementations deal with it? Also, "theURI" and "anURI" are not good names